### PR TITLE
Consistent variable naming, sorting of members in classes and general…

### PR DESCRIPTION
**Pull request by agneborn98**
… beautification

### CODE REVIEW

File: src/managers/ConfigManager.cpp [hunk 1]
```diff
@@ -1,103 +1,97 @@
...
+                {"columns",    required_argument, 0, 'c'},
                 {"timestamps", required_argument, 0, 't'},
-                {"output", required_argument, 0, 'o'},
-                {"text", required_argument, 0, 'x'},
-                {"inputFile", required_argument, 0, 'i'},
-                {0, 0, 0, 0}};
+                {"output",     required_argument, 0, 'o'},
+                {"text",       required_argument, 0, 'x'},
+                {"inputFile",  required_argument, 0, 'i'},
+                {0, 0,                            0, 0}};
```
Comment (karljohan12) [inline src/managers/ConfigManager.cpp]: Skal den anden kolonne i nederste række ikke aligne ligesom resten

File: src/managers/ModelManager.cpp [hunk 2]
```diff
@@ -2,129 +2,160 @@
...
+    } else if (container.text != value) {
         //TODO: Emit text model to MQTT HERE
         container.reCheck = true;
     }
 }
 
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "misc-no-recursion"
+
 //Recursive chain call is on purpose
```
Comment (karljohan12) [inline src/managers/ModelManager.cpp]: remove

File: src/managers/ModelManager.cpp [hunk 3]
```diff
@@ -2,129 +2,160 @@
...
-        timeSeries.emplace_back(column.error, column.isAbsolute, count, column.col);
+    for (auto &column: timeSeriesConfig) {
+        timeSeries.emplace_back(column.error, column.isAbsolute, count,
+                                column.col);
         count++;
     }
     count = 0;
-    for (auto &textColumn : text_cols){
+    for (auto &textColumn: text_cols) {
         textModels.emplace_back(count, textColumn);
```
Comment (karljohan12) [inline src/managers/ModelManager.cpp]: text_cols

File: src/models/Gorilla.cpp [hunk 4]
```diff
@@ -2,212 +2,180 @@
...
-                value_xor_last_value >> last_trailing_zero_bits, 
-                meaningful_bits
+        uint8_t leading_zero_bits = leadingZeros(value_xor_last_value);
+        uint8_t trailing_zero_bits = trailingZeros(value_xor_last_value);
+        appendAOneBit(&compressedValues);
+
+        if (leading_zero_bits >= lastLeadingZeroBits
+            && trailing_zero_bits >= lastTrailingZeroBits) {
+            appendAZeroBit(&compressedValues);
+            uint8_t meaningful_bits = VALUE_SIZE_IN_BITS
```
Comment (karljohan12) [inline src/models/Gorilla.cpp]: _

File: src/models/Gorilla.cpp [hunk 5]
```diff
@@ -2,212 +2,180 @@
...
-        {
-            append_a_zero_bit(&compressed_values);
-            uint8_t meaningful_bits = VALUE_SIZE_IN_BITS 
-                - last_leading_zero_bits 
-                - last_trailing_zero_bits;
-            append_bits(&compressed_values, 
-                value_xor_last_value >> last_trailing_zero_bits, 
-                meaningful_bits
+        uint8_t leading_zero_bits = leadingZeros(value_xor_last_value);
+        uint8_t trailing_zero_bits = trailingZeros(value_xor_last_value);
```
Comment (karljohan12) [inline src/models/Gorilla.cpp]: __

File: src/models/Gorilla.cpp [hunk 6]
```diff
@@ -2,212 +2,180 @@
...
+
+        if (leading_zero_bits >= lastLeadingZeroBits
+            && trailing_zero_bits >= lastTrailingZeroBits) {
+            appendAZeroBit(&compressedValues);
+            uint8_t meaningful_bits = VALUE_SIZE_IN_BITS
+                                      - lastLeadingZeroBits
+                                      - lastTrailingZeroBits;
+            appendBits(&compressedValues,
+                       value_xor_last_value >> lastTrailingZeroBits,
+                       meaningful_bits
```
Comment (karljohan12) [inline src/models/Gorilla.cpp]: __

File: src/models/Gorilla.cpp [hunk 7]
```diff
@@ -2,212 +2,180 @@
...
+    while (_number_of_bits > 0) {
         uint8_t bits_written;
 
-        if(_number_of_bits > data->remaining_bits){
-            uint8_t shift = _number_of_bits -  data->remaining_bits;
-            data -> current_byte |= (uint8_t)((bits >> shift) & ((1 << data->remaining_bits) - 1));
-            bits_written = data->remaining_bits;
+        if (_number_of_bits > data->remainingBits) {
+            uint8_t shift = _number_of_bits - data->remainingBits;
+            data->currentByte |= (uint8_t) ((bits >> shift) &
```
Comment (karljohan12) [inline src/models/Gorilla.cpp]: __

File: src/models/PMCMean.cpp [hunk 8]
```diff
@@ -7,150 +7,151 @@
 #include <iostream>
 
 PmcMean::PmcMean(double &error_bound, bool error_absolute)
-        : error(error_bound){
+        : error(error_bound) {
```
Comment (karljohan12) [inline src/models/PMCMean.cpp]: __

File: src/models/PMCMean.cpp [hunk 9]
```diff
@@ -7,150 +7,151 @@
...
 
-    if(is_value_within_error_bound(next_min_value, average) && is_value_within_error_bound(next_max_value, average)){
-        min_value = next_min_value;
-        max_value = next_max_value;
-        sum_of_values = next_sum_of_values;
+    if (isValueWithinErrorBound(next_min_value, average) &&
+        isValueWithinErrorBound(next_max_value, average)) {
+        minValue = next_min_value;
+        maxValue = next_max_value;
+        sumOfValues = next_sum_of_values;
```
Comment (karljohan12) [inline src/models/PMCMean.cpp]: ___

File: src/models/PMCMean.cpp [hunk 10]
```diff
@@ -7,150 +7,151 @@
...
         return 0;
     }
 }
 
-int PmcMean::is_value_within_error_bound(float real_value, float approx_value){
-    if(equal_or_nan_pmc(real_value, approx_value)){
+int
+PmcMean::isValueWithinErrorBound(float real_value, float approx_value) const {
+    if (equalOrNanPmc(real_value, approx_value)) {
         return 1;
```
Comment (karljohan12) [inline src/models/PMCMean.cpp]: __

File: src/models/PMCMean.cpp [hunk 11]
```diff
@@ -7,150 +7,151 @@
...
-  sum_of_values = 0;
-  length = 0;
-}
-
-std::vector<float> PmcMean::grid_pmc_mean(float value, int timestamp_count){
+std::vector<float> PmcMean::gridPmcMean(float value, int timestamp_count) {
     std::vector<float> result;
-    for(int i = 0; i < timestamp_count; i++){
+    for (int i = 0; i < timestamp_count; i++) {
         result.push_back(value);
```
Comment (karljohan12) [inline src/models/PMCMean.cpp]: __

File: src/models/PMCMean.cpp [hunk 12]
```diff
@@ -7,150 +7,151 @@
...
+#pragma clang diagnostic pop
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Woverloaded-shift-op-parentheses"
+
+TEST_CASE("Grid") {
 
     //Grid
     double error_bound = 1;
     PmcMean p(error_bound, true);
```
Comment (karljohan12) [inline src/models/PMCMean.cpp]: _

File: src/models/PMCMean.cpp [hunk 13]
```diff
@@ -7,150 +7,151 @@
...
     }
 
-    auto res = p.grid_pmc_mean(p.get_sum_of_values()/p.get_length(), vals.size());
+    auto res = p.gridPmcMean(p.sumOfValues / p.length, vals.size());
     bool equal = true;
-    for(int i = 0; i < vals.size(); i++){
-        if(std::fabs(vals[i]-res[i]) > error_bound){
+    for (int i = 0; i < vals.size(); i++) {
+        if (std::fabs(vals[i] - res[i]) > error_bound) {
             equal = false;
```
Comment (karljohan12) [inline src/models/PMCMean.cpp]: __

File: src/models/PMCMean.h [hunk 14]
```diff
@@ -5,32 +5,28 @@
...
-    int lastTimestamp;
+    [[nodiscard]] float getBytesPerValue() const;
+
+    PmcMean &operator=(const PmcMean &instance);
+
+private:
+    bool isErrorAbsolute;
+
+    int isValueWithinErrorBound(float real_value, float approx_value) const;
+
```
Comment (karljohan12) [inline src/models/PMCMean.h]: __

File: src/models/Swing.h [hunk 15]
```diff
@@ -7,70 +7,55 @@
...
+    double lowerBoundSlope;
+    double lowerBoundIntercept;
     int length;
-    bool error_absolute;
+    bool errorAbsolute;
+    bool bothLatLongSeen;
 
-    int isNan(double val);
-    int equalOrNAN(double v1, double v2);
+    Swing(double &error_bound, bool is_error_absolute);
```
Comment (karljohan12) [inline src/models/Swing.h]: __

File: src/models/Swing.h [hunk 16]
```diff
@@ -7,70 +7,55 @@
...
+            long firstTimestamp,
+            double firstValue,
+            long finalTimestamp,
+            double finalValue);
+
+    slopeAndIntercept
+    decodeAndComputeSlopeAndIntercept(long firstTimestamp, long lastTimestamp,
+                                      double min_value, double maxValue,
+                                      int value);
+
```
Comment (karljohan12) [inline src/models/Swing.h]: __

**Review by karljohan12 — CHANGES_REQUESTED** [review]

File: src/managers/ConfigManager.cpp (hunk 1 above)
Comment (agneborn98) [reply src/managers/ConfigManager.cpp]: Fixed.

File: src/managers/ModelManager.cpp (hunk 2 above)
Comment (agneborn98) [reply src/managers/ModelManager.cpp]: Fixed.

File: src/managers/ModelManager.cpp (hunk 3 above)
Comment (agneborn98) [reply src/managers/ModelManager.cpp]: Fixed.

File: src/models/Gorilla.cpp (hunk 4 above)
Comment (agneborn98) [reply src/models/Gorilla.cpp]: Fixed.

File: src/models/Gorilla.cpp (hunk 5 above)
Comment (agneborn98) [reply src/models/Gorilla.cpp]: Fixed.

File: src/models/Gorilla.cpp (hunk 6 above)
Comment (agneborn98) [reply src/models/Gorilla.cpp]: Fixed.

File: src/models/Gorilla.cpp (hunk 7 above)
Comment (agneborn98) [reply src/models/Gorilla.cpp]: Fixed.

File: src/models/PMCMean.cpp (hunk 8 above)
Comment (agneborn98) [reply src/models/PMCMean.cpp]: Fixed.

File: src/models/PMCMean.cpp (hunk 9 above)
Comment (agneborn98) [reply src/models/PMCMean.cpp]: Fixed.

File: src/models/PMCMean.cpp (hunk 10 above)
Comment (agneborn98) [reply src/models/PMCMean.cpp]: Fixed.

File: src/models/PMCMean.cpp (hunk 11 above)
Comment (agneborn98) [reply src/models/PMCMean.cpp]: Fixed.

File: src/models/PMCMean.cpp (hunk 12 above)
Comment (agneborn98) [reply src/models/PMCMean.cpp]: Fixed.

File: src/models/PMCMean.cpp (hunk 13 above)
Comment (agneborn98) [reply src/models/PMCMean.cpp]: Fixed.

File: src/models/PMCMean.h (hunk 14 above)
Comment (agneborn98) [reply src/models/PMCMean.h]: Fixed.

File: src/models/Swing.h (hunk 15 above)
Comment (agneborn98) [reply src/models/Swing.h]: Fixed.

File: src/models/Swing.h (hunk 16 above)
Comment (agneborn98) [reply src/models/Swing.h]: Fixed.

File: src/models/Gorilla.cpp [hunk 17]
```diff
@@ -2,212 +2,179 @@
...
 
-uint32_t Gorilla::float_to_bit(float val){
-    return *(uint32_t *)&val;
+uint32_t Gorilla::floatToBit(float val) {
+    return *(uint32_t *) &val;
 }
 
+void Gorilla::fitValueGorilla(float value) {
+    uint32_t value_as_integer = floatToBit(
+            value); // Read the binary representation of the float value as an integer, which can then be used for bitwise operations.
```
Comment (karljohan12) [inline src/models/Gorilla.cpp:50]: __

File: src/managers/ConfigManager.cpp [hunk 18]
```diff
@@ -1,103 +1,98 @@
...
+                {"columns",    required_argument, 0, 'c'},
                 {"timestamps", required_argument, 0, 't'},
-                {"output", required_argument, 0, 'o'},
-                {"text", required_argument, 0, 'x'},
-                {"inputFile", required_argument, 0, 'i'},
-                {0, 0, 0, 0}};
+                {"output",     required_argument, 0, 'o'},
+                {"text",       required_argument, 0, 'x'},
+                {"inputFile",  required_argument, 0, 'i'},
+                {0          ,  0                , 0,  0 }
```
Comment (teisvh) [inline src/managers/ConfigManager.cpp]: kommaer i linje 51 følger ikke det samme mønster som foregående linjer

Comment (agneborn98) [reply src/managers/ConfigManager.cpp]: Fixed.

File: src/managers/ReaderManager.cpp [hunk 19]
```diff
@@ -7,26 +7,34 @@
...
-    this->csvFileStream.open(this->configManager.getInputFile()/*"../Cobham_hour.csv"*/, std::ios::in);
+        : configManager(configFile), timestampManager(configManager),
+          modelManager(configManager.timeseriesCols, configManager.textCols,
+                       timestampManager) {
+    this->csvFileStream.open(
+            "../" + this->configManager.inputFile/*"../Cobham_hour.csv"*/,
+            std::ios::in);
 
-    bothLatLongSeen = false;
+    auto bothLatLongSeen = false;
```
Comment (teisvh) [inline src/managers/ReaderManager.cpp]: ??

Comment (teisvh) [reply src/managers/ReaderManager.cpp]: ikke sikker på, at det her er rigtigt

**Review by teisvh — APPROVED** [review]

File: src/managers/ReaderManager.cpp (hunk 19 above)
Comment (agneborn98) [reply src/managers/ReaderManager.cpp]: Fixed.

File: src/models/Gorilla.cpp (hunk 17 above)
Comment (agneborn98) [reply src/models/Gorilla.cpp:50]: Fixed.

**Review by karljohan12 — APPROVED** [review]

**Pull request merged**